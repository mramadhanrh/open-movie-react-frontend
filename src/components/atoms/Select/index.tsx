import styled from 'styled-components';

const Select = styled.select`
  border: none;
  box-shadow: 0px 0px 0px 0px #f86f22;
  border-radius: 3px;
  background-color: ${({ theme }) => theme.color.mischka};
  padding: 10px;
  font-weight: 500;
  cursor: pointer;

  ::placeholder {
    font-weight: normal;
  }
`;

export const Option = styled.option`
  cursor: pointer;
  padding: 10px;
`;

export default Select;
