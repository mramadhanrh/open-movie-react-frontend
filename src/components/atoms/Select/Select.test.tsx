import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ThemeProvider } from 'styled-components';
import Select, { Option } from '.';
import defaultTheme from '../../../constants/theme';

describe('Select', () => {
  const selectComp = (
    <ThemeProvider theme={defaultTheme}>
      <Select defaultValue={1}>
        <Option value={1}>Value 1</Option>
        <Option value={2}>Value 2</Option>
        <Option value={3}>Value 3</Option>
      </Select>
    </ThemeProvider>
  );

  test('match snapshot', () => {
    expect(toJson(shallow(selectComp))).toMatchSnapshot();
  });

  test('can select item', () => {
    const wrapper = mount(selectComp);
    expect(wrapper.getDOMNode<HTMLSelectElement>().value).toBe('1');
    wrapper.getDOMNode<HTMLSelectElement>().value = '2';
    wrapper.simulate('change');
    expect(wrapper.getDOMNode<HTMLSelectElement>().value).toBe('2');
    wrapper.getDOMNode<HTMLSelectElement>().value = '3';
    wrapper.simulate('change');
    expect(wrapper.getDOMNode<HTMLSelectElement>().value).toBe('3');
  });
});
