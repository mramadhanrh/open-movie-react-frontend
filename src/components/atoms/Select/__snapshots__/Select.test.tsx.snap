// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "buttonColor": Object {
        "primary": "#754600",
        "secondary": "#F86F22",
        "text": "transparent",
      },
      "buttonTextColor": Object {
        "primary": "#FFFFFF",
        "secondary": "#000000",
        "text": "#000000",
      },
      "color": Object {
        "athensGray": "#E6E8EC",
        "black": "#000000",
        "bridalHeath": "#FFF9F0",
        "chelseaGem": "#A46200",
        "coal": "#2B2B2B",
        "ecstasy": "#F86F22",
        "mischka": "#D7D8DE",
        "pizzaz": "#FB8C00",
        "primary": "#754600",
        "sandyBeach": "#FFEACA",
        "secondary": "#F86F22",
        "tussock": "#BB883C",
        "white": "#FFFFFF",
      },
      "textColor": Object {
        "primary": "#754600",
        "secondary": "#F86F22",
        "tertiary": "#2B2B2B",
      },
    }
  }
>
  <styled.select
    defaultValue={1}
  >
    <styled.option
      value={1}
    >
      Value 1
    </styled.option>
    <styled.option
      value={2}
    >
      Value 2
    </styled.option>
    <styled.option
      value={3}
    >
      Value 3
    </styled.option>
  </styled.select>
</ContextProvider>
`;
